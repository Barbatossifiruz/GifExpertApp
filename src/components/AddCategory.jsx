import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState("");

    //! Validacion to lower case
    const search = inputValue.toLowerCase();

    //! Capitalizer

    //* V1 solo en mayuscula la primera letra
    // const CapFirsLetter = (str) => {
    //     return str.charAt(0).toUpperCase() + str.slice(1);
    // };

    //* V2 Capitalizar con RegEx
    const TittleRegEx = inputValue.replace(/(^\w{1})|(\s+\w{1})/g, (letra) => letra.toUpperCase());

    //! Constante en cambio (Buscador)
    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    //! Informacion de (Buscador)
    const onSubmit = (event) => {
        //! Validacion en Buscador
        event.preventDefault();
        if (search.trim().length <= 1) return;

        setInputValue("");
        onNewCategory(search.trim());

        //!Muestra el resultado Capitalizado
        onNewCategory(TittleRegEx);
    };

    return (
        <>
            <form onSubmit={onSubmit}>
                <input type="text" placeholder="Buscar Gif" value={inputValue} onChange={onInputChange} />
            </form>
        </>
    );
};
