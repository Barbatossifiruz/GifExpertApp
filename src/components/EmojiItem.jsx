export const EmojiItem = ({ title, url }) => {
    return (
        <>
            <div className="w-full">
                <img className="emoji" src={url} alt={title} />
            </div>
        </>
    );
};
