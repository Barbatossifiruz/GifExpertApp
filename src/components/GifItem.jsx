export const GifItem = ({ title, url }) => {
    return (
        <>
            <div className="max-w-sm bg-white border-gray-200 rounded-lg   dark:bg-[#1f1f1f] drop-shadow-xl ">
                <img className="rounded-t-lg " src={url} alt={title} />
                <h5 className="font-bold text-md text-center text-white p-2">{title}</h5>
            </div>
        </>
    );
};
