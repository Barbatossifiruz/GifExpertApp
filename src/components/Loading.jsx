//!Loading section

export const Loading = () => {
    Loading && <h2>Cargando...</h2>;
};
