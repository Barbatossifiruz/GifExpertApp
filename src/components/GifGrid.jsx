/*
//! Orden de importaciones
 *1 - React
 *2- Importaciones de terceros
 *3- Nuestro codigo
 *4- Funciones
*/

import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { Loading } from "./Loading";

export const GifGrid = ({ category }) => {
    //! Loading
    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            <Loading isLoading={isLoading} />

            <div className="card-grid">
                {images.map((image) => (
                    <GifItem key={image.id} {...image} />
                ))}
            </div>
        </>
    );
};
