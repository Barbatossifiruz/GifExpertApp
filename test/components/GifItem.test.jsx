/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en <GitItem/>", () => {
    const title = "saitama";
    const url = "https://one-punch.com/saitama.jpg";

    test("Debe hacer match con el snapshot", () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });
});
