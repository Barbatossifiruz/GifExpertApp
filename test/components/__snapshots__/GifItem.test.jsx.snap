// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="max-w-sm bg-white border-gray-200 rounded-lg   dark:bg-[#1f1f1f] drop-shadow-xl "
  >
    <img
      alt="saitama"
      class="rounded-t-lg "
      src="https://one-punch.com/saitama.jpg"
    />
    <h5
      class="font-bold text-md text-center text-white p-2"
    >
      saitama
    </h5>
  </div>
</div>
`;
